invokeai/app/cli_app.py

```diff
@@ -129,11 +129,7 @@ def invoke_cli():
     args = Args()
     config = args.parse_args()
 
-    generate = get_generate(args, config)
-
-    # NOTE: load model on first use, uncomment to load at startup
-    # TODO: Make this a config option?
-    # generate.load_model()
+    model_manager = get_model_manager(args, config)
```

Comment:
Same comment as in the API about construction

Reply:
As noted earlier, there is actually a lot of initialization and arg processing going on in `get_model_manager()` and it's not clear to me what is needed to create a new Service. Looking at `InvocationServies` I see lots of different types of object being registered as services and they don't seem to share a common base class?

If nothing else needs to be done to make ModelManager into a service, then I think it's done.

Reply:
A service needs an abstract base class to define the interface (so other implementations can be provided). I think I was mostly hoping that construction could happen in the init method for the model manager, and if there were things that could be split up that we'd introduce additional services.

Reply:
I'm happy to do (or contribute) to this in the next phase of the refactor. We'll need an abstract base class for the Model Manager and the reconstruction/upscaling models. With respect to the latter, I think that maybe a PostProcessor class that takes an image in and produces an image out is the right paradigm?